hushline/settings/__init__.py

```diff
@@ -560,6 +565,136 @@ def advanced() -> str:
+            ),
+        )
+
+        guidance_prompt_values = OrganizationSetting.fetch_one(OrganizationSetting.GUIDANCE_PROMPTS)
+        if guidance_prompt_values is None:
+            guidance_prompt_values = []
+        user_guidance_prompt_forms = [
+            UserGuidancePromptContentForm(
+                heading_text=guidance_prompt_values[i].get("heading_text", ""),
+                prompt_text=guidance_prompt_values[i].get("prompt_text", ""),
+            )
+            for i in range(len(guidance_prompt_values))
+        ]
+
+        user_guidance_add_prompt_form = UserGuidanceAddPromptForm()
+
+        status_code = 200
+        if request.method == "POST":
+            # Show user guidance form
+            if (user_guidance_form.submit.name in request.form) and user_guidance_form.validate():
+                OrganizationSetting.upsert(
+                    key=OrganizationSetting.GUIDANCE_ENABLED,
+                    value=user_guidance_form.show_user_guidance.data,
+                )
+                if user_guidance_form.show_user_guidance.data:
+                    show_user_guidance = True
+                    flash("👍 User guidance enabled.")
+                else:
+                    show_user_guidance = False
+                    flash("👍 User guidance disabled.")
+                return redirect(url_for(".guidance"))
+
+            # Emergency exit form
+            elif (
+                user_guidance_emergency_exit_form.submit.name in request.form
+            ) and user_guidance_emergency_exit_form.validate():
+                OrganizationSetting.upsert(
+                    key=OrganizationSetting.GUIDANCE_EXIT_BUTTON_TEXT,
+                    value=user_guidance_emergency_exit_form.exit_button_text.data,
+                )
+                OrganizationSetting.upsert(
+                    key=OrganizationSetting.GUIDANCE_EXIT_BUTTON_LINK,
+                    value=user_guidance_emergency_exit_form.exit_button_link.data,
+                )
+                flash("👍 Emergency exit button updated successfully.")
+                return redirect(url_for(".guidance"))
+
+            # Add prompt form
+            elif (
+                user_guidance_add_prompt_form.submit.name in request.form
+            ) and user_guidance_add_prompt_form.validate():
+                new_prompt_value = {
+                    "heading_text": "",
+                    "prompt_text": "",
+                }
+                guidance_prompt_values.append(new_prompt_value)
+                user_guidance_prompt_forms.append(UserGuidancePromptContentForm())
+
+                OrganizationSetting.upsert(
+                    key=OrganizationSetting.GUIDANCE_PROMPTS,
+                    value=guidance_prompt_values,
+                )
+
+                flash("👍 Prompt added.")
+                return redirect(url_for(".guidance"))
+
+            # Guidance prompt forms
+            else:
+                form_submitted = False
```

Comment:
This logic is slightly convoluted to me. Can you add a brief comment? I feel like there should be a simpler way to do this, but I'm not seeing it atm